### PR TITLE
Fix MIPS asm parser to display vars in disasm ##disasm

### PR DESCRIPTION
**Pull request by sivaramaaa**
* **Before**

mips_before

* **After**

mips_after

### CODE REVIEW

**Comment by radare** [comment]
which other archs are missing this? i think we only have x86, arm, arm64... and now mips, right?

will be good to extend this to other archs, any proposal?

**Comment by Maijin** [comment]
AVR has an issue open https://github.com/radare/radare2/issues/11468

**Comment by radare** [comment]
Looks green why its wip?

File: libr/parse/p/parse_mips_pseudo.c [hunk 1]
```diff
@@ -251,116 +251,85 @@ static int parse(RParse *p, const char *data, char *str) {
...
 	}
-	vars = p->varlist (p->anal, f, 'v');
-	args = p->varlist (p->anal, f, 'a');
-	spargs = p->varlist (p->anal, f, 'e');
-	/*iterate over stack pointer arguments/variables*/
-	r_list_foreach (spargs, spiter,sparg) {
-		if (sparg->delta < 10) {
-			snprintf (oldstr, sizeof (oldstr)-1, "[%s + %d]",
-				p->anal->reg->name[R_REG_NAME_SP], sparg->delta);
+	bpargs = p->varlist (p->anal, f, 'b');
```
Comment (radare) [inline libr/parse/p/parse_mips_pseudo.c]: define and assign in the same line

File: libr/parse/p/parse_mips_pseudo.c [hunk 2]
```diff
@@ -251,116 +251,85 @@ static int parse(RParse *p, const char *data, char *str) {
...
+	spargs = p->varlist (p->anal, f, 's');
+	bool ucase = IS_UPPER (*tstr);
+	r_list_foreach (spargs, iter, var) {
+		char *tmpf = NULL;
```
Comment (radare) [inline libr/parse/p/parse_mips_pseudo.c]: no need to initialize, all paths set this var

File: libr/parse/p/parse_mips_pseudo.c [hunk 3]
```diff
@@ -251,116 +251,85 @@ static int parse(RParse *p, const char *data, char *str) {
...
-	args = p->varlist (p->anal, f, 'a');
-	spargs = p->varlist (p->anal, f, 'e');
-	/*iterate over stack pointer arguments/variables*/
-	r_list_foreach (spargs, spiter,sparg) {
-		if (sparg->delta < 10) {
-			snprintf (oldstr, sizeof (oldstr)-1, "[%s + %d]",
-				p->anal->reg->name[R_REG_NAME_SP], sparg->delta);
+	bpargs = p->varlist (p->anal, f, 'b');
+	spargs = p->varlist (p->anal, f, 's');
+	bool ucase = IS_UPPER (*tstr);
```
Comment (radare) [inline libr/parse/p/parse_mips_pseudo.c]: const

File: libr/parse/p/parse_mips_pseudo.c [hunk 4]
```diff
@@ -251,116 +251,85 @@ static int parse(RParse *p, const char *data, char *str) {
...
+                        char *comma = strchr (oldstr, ',');
+                        if (comma) {
+                                *comma = 0;
+                                r_str_case (oldstr, true);
+                                *comma = ',';
+                        }
+                }
+                if (strstr (tstr, oldstr)) {
+                        if (p->localvar_only) {
+                                newstr = r_str_newf ("(%s)", var->name);
```
Comment (radare) [inline libr/parse/p/parse_mips_pseudo.c]: maybe newstr should be local here and using ternary operator will  look better.

**Pull request merged**